### ISSUE TITLE
Extend feedback flexibility to all question types

### ISSUE THREAD

**Issue by sacole167** [isle-project/isle-editor#1252]
Would be good to extent the incremental feedback capability to all question types (this is a high priority for me)

Less important: in the long run, you might want to consider allowing users to set the number of submissions they want to allow, rather than just allowing 1 or infinite submissions.

**Comment by Planeshifter** [comment]
For the number question, I added a new `nTries` attribute this morning that allows one to specify the number of incorrect submissions before the correct answer is supplied. You can see this in action here:

https://isle.stat.cmu.edu/demonstrations/multiple-tries/

Code:
```
<NumberQuestion
  question="What is 5+3?"
  nTries={5} provideFeedback={true}
  solution={8}
/>
```

For the number question, I am not convinced of an additional need to allow infinite submissions. One could just supply a large number for the number of tries.

**Comment by sacole167** [comment]
I thought Xiaoyi told me that provideFeedback=incremental means the student has infinite tries.

As an instructor, I have never felt the need to infinite tries. I have sometimes given students 6 tries (which seems like close to infinite to me ; - ) ). But I was just a teaching discussion this week where the instructor said he allowed infinite tries. So perhaps it is something people use.

I forgot to mention: you may also want to consider a text field for feedback, e.g., "If you got this wrong, try thinking about the problem as a fraction and try again." As you may know, Canvas allows this and also allows specific feedback for specific wrong solutions (e.g., in MultipleChoiceQuestions). That is not something I use very much myself--too much work!--but some people might.

**Comment by Planeshifter** [comment]
Yes, for the multiple-choice question that's the case. There `provideFeedback` can be `full`, `none`, or `incremental`. Since the space of possible solutions is so limited, students can just pick one answer after the other and then will reach the result eventually. On the other hand, for say the numeric questions that's not the case and thus I think it's prudent to limit the number of tries instead. 

For question types aside from `<MultipleChoiceQuestion />`,  `provideFeedback` is right now just a boolean that is either `true` or `false`.  

Some of the questions allow one to tweak the message displayed upon a correct or incorrect answer, which can be used to supply feedback. When adding options for multiple number of tries, I will also look to bring all the question types to feature parity.

**Comment by sacole167** [comment]
It seems to give the correct answer on the first try:

<img width="1340" alt="Screen Shot 2020-08-31 at 2 09 47 PM" src="https://user-images.githubusercontent.com/69658229/91769214-ccfbdf00-eb93-11ea-8866-5e1e2c59e690.png">

**Comment by sacole167** [comment]
This problem is fixed by updating the editor.

Incidentally, there may be use-cases where it is desired to use nTries when provideFeedback is set to {false}.

**Comment by sacole167** [comment]
nTries still not available on multiple choice question, right?

**Comment by Planeshifter** [comment]
Not yet, but I could definitely make this a priority. Could you please give me an example where having `nTries` would be beneficial for multiple choice questions and where the `full` and `incremental` options for `provideFeedback` are not sufficient?

**Comment by sacole167** [comment]
I see. Incremental is good enough for my purposes. Perhaps some more explanation of what Incremental does on the [documentation page](https://isledocs.com/docs/multiple-choice-question) and the Configurator menu would be helpful

**Comment by Planeshifter** [comment]
Aw, I see that the description does not accurately describe that `incremental` coincides with multiple tries until the correct solution is reached. Will definitely update the docs!

Should you run into a situation where `nTries` would be useful to have, please let me know. Trying to balance the components' usefulness and simplicity by adding not too many options.
